### DISCUSSION TITLE
Example/sample firmware for "b-parasite" open-source soil moisture sensor?

### DISCUSSION THREAD

**Discussion by Hedda** [ffenix113/zigbee_home#5] [Ideas]
Please consider building a proof-of-concept firmware samples/examples for the **b-parasite** (hardware-v2) DIY and open-source soil moisture + ambient temperature/humidity/light (plant/flower) sensor by rbaron, can be built on either [nRF52840](https://www.nordicsemi.com/products/nrf52840) and [nRF52833](https://www.nordicsemi.com/products/nrf52833) Zigbee radio modules:

https://github.com/rbaron/b-parasite

Upstream discussion:

* https://github.com/rbaron/b-parasite/discussions/178

Hardware revision 2.0.0:

https://github.com/rbaron/b-parasite/tree/hardware-v2

https://github.com/rbaron/b-parasite/wiki/Hardware-Versions#version-20x

This is an open source hardware project is a PCB design but you can also build your own using any capacitive soil moisture sensors, see:

https://rbaron.net/blog/2021/04/05/How-capacitive-soil-moisture-sensors-work.html

FYI, as that is an open source there are also other forks/projects based on it, including this that can be purchased as a complete kit

https://github.com/stanvn/zigbee-plant-sensor

https://homeplantsensor.com/

Upstream repo also includes a "blinky" sample as a classic "Hello, world" as well as a few other samples:

https://github.com/rbaron/b-parasite/tree/main/code/nrf-connect/samples/

By the way, rbaron has merged that Zigbee sample to b-parasite upstream repository and he has tested it with Home Assistant’s ZHA integration (which works without modifications) as well as with Zigbee2MQTT (which requires zigbee-herdsman converter b-parasite.js from his [PR](https://github.com/rbaron/b-parasite/pull/81) that needs to be added as per [Z2M instructions to support new devices](https://www.zigbee2mqtt.io/advanced/support-new-devices/01_support_new_devices.html)):

https://github.com/rbaron/b-parasite/pull/81

https://github.com/rbaron/b-parasite/blob/hardware-v2/code/nrf-connect/samples/zigbee

https://github.com/rbaron/b-parasite/blob/hardware-v2/code/nrf-connect/samples/zigbee/README.md

PS: That and similar hardware for Zigbee flower and plant sensors for soil moisture + light + temperature + humidity discussed here:

https://community.home-assistant.io/t/zigbee-flower-and-plant-sensors-for-soil-moisture-light-temperature-humidity/272803/

**Comment by Hedda** [comment]
> FYI, as that is an open source there are also other forks/projects based on it, including this that can be purchased as a complete kit
> 
> https://github.com/stanvn/zigbee-plant-sensor
> 
> https://homeplantsensor.com/

Another FYI, there is also this similar semi-closed "EFEKTA" project based on nRF52840 which believe is also based on Nordic Semi nRF5 SDK:

https://github.com/smartboxchannel/Plant-Watering-Sensor-Zigbee

https://hackaday.io/project/181075-efekta-pws-zigbee

https://www.tindie.com/products/diyberk/plant-watering-sensor-efekta-zigbee/


https://github.com/smartboxchannel/EFEKTA_E73C_PWS_MODULE

https://github.com/smartboxchannel/EFEKTA_E73C_PWS_MODULE/tree/master/Zigbee

https://www.youtube.com/watch?v=LTzc0X-ESm4&feature=youtu.be&ab_channel=AndrewEfektaSB

![image](https://github.com/ffenix113/zigbee_home/assets/6320001/9ded9e8d-dd12-4679-9dae-4ad72f10b5f9)

His prototypes used a common type of basic "capacitive soil moisture sensor" boards sold in China were originally designed for Arduino

https://www.openhardware.io/view/767/Wireless-Module-for-Capacitive-Soil-Moisture-Sensor-v20

![image](https://github.com/ffenix113/zigbee_home/assets/6320001/366f0b35-d453-44ed-b782-920ee46fb773)

**Reply by ffenix113** [reply]
I've bought myself the exact one sensor that is in your last screenshot, will work on the support as will have time.

**Comment by Hedda** [comment]
Both projects were inspired by DIYRuZ modkam.ru community who designed many Zigbee DIY hardware based on CC2530 mentioned here:

https://github.com/zigpy/zigpy/discussions/646

**Comment by ffenix113** [comment]
Thank you for the suggestion.

I will keep this in mind, but currently my access to sensors is limited. I will add sensors(and necessary zigbee clusters) as soon as I would have them.

I would check to see if it would be possible to "simulate" this moisture sensor with something else, as this will help to add support faster.

**Comment by ffenix113** [comment]
Hello @Hedda !

I've just merged the ADC implementation, and the required sensors. Latest-dev version of zigbee2mqtt also now supports soil moisture sensor.
For [Capacitive Soil Moisture Sensor v2.0](https://www.openhardware.io/view/767/Wireless-Module-for-Capacitive-Soil-Moisture-Sensor-v20) the configuration could be as follows:
```yaml
   - type: soil_moisture_adc
     adc_pin: 
       pin: 0.04
       oversampling: 4
     # This values are figured out experimentally with ADC configuration above.
     # They could be different for other ADC setup or sensor.
     min_moisture_mv: 730
     max_moisture_mv: 430
```

I will leave this open until better configuration is available on sensors and configuration.
